File: BreakOut3b/BOMovingObject.m
//
//  BOMovingObject.m
//  BreakOut3b
//
//  Created by Carl Tengwall on 2012-11-06.
//  Copyright (c) 2012 Carl Tengwall. All rights reserved.
//

#import "BOMovingObject.h"

@interface BOMovingObject() {
    float speed;                // Keep the speed private
}
@property float speed;
@end

@implementation BOMovingObject

@synthesize speed = _speed;
@synthesize direction = _direction;
@synthesize position = _position;
@synthesize velocity = _velocity;


- (id)initAtPosition: (CGPoint) pos
{
    self = [super init];
    if (self) {
        _position = pos;
        _speed = MO_SPEED;
        _direction = CGPointMake(MO_DIR_X,MO_DIR_Y);
        _velocity = CGPointMake(_speed*_direction.x, _speed*_direction.y);
    }
    return self;
}

@end

File: BreakOut3b/BOMovingObject.swift
import CoreGraphics
import Foundation

let MO_SIZE: CGFloat = 10
let MO_SPEED: CGFloat = 400
let MO_POS_X: CGFloat = 180
let MO_POS_Y: CGFloat = 220
let MO_DIR_X: CGFloat = 1
let MO_DIR_Y: CGFloat = -1

@objc(BOMovingObject)
final class BOMovingObject: NSObject {
    @objc var direction: CGPoint
    @objc var position: CGPoint
    @objc var velocity: CGPoint

    private let speed: CGFloat

    @objc(initAtPosition:)
    init(position: CGPoint) {
        self.position = position
        self.speed = MO_SPEED
        self.direction = CGPoint(x: MO_DIR_X, y: MO_DIR_Y)
        self.velocity = CGPoint(x: speed * direction.x, y: speed * direction.y)
        super.init()
    }
}
